### PR TITLE
Optimized steal item searcher to not search through the entire game world.

### PR DESCRIPTION
**Pull request by Watermelon914**
## About The Pull Request

See title. It now overrides procs on the items for special handling instead. This is more performant than a for loop iterating through the world.

## Why It's Good For The Game

More performant

### CODE REVIEW

File: code/game/gamemodes/objective_items.dm [hunk 1]
```diff
@@ -309,36 +389,57 @@
...
+	GLOB.steal_item_handler.objectives_by_path[/obj/item/pinpointer/nuke] += src
+
 /datum/objective_item/special/aegun
 	name = "an advanced energy gun"
 	targetitem = /obj/item/gun/energy/e_gun/nuclear
 	difficulty = 10
 
+/obj/item/gun/energy/e_gun/nuclear/add_stealing_item_objective()
+	GLOB.steal_item_handler.objectives_by_path[/obj/item/gun/energy/e_gun/nuclear] += src
+
```
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm:402]: almost all of these ones below seem unfeasible to do, as they are created midround, and not maploaded

**Comment by Iamgoofball** [comment]
performance issue created by you, as per ruling by maintainers, tagging w/ GBP no update

**Comment by Watermelon914** [comment]
why add any label at all?

File: code/game/gamemodes/objective_items.dm (hunk 1 above)
Comment (Watermelon914) [reply code/game/gamemodes/objective_items.dm:402]: it's there just in case, They won't be considered anyways unless exists_on_map is set to true

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Can you change the C+P to a proc on `/obj/item` instead? One that still takes `type` as an argument in the case of subtypes?

**Comment by LemonInTheDark** [comment]
This is actually quite important, because the size of that for loop iteration was triggering byond's infinite loop detection. s not a good false positive to have

**Pull request merged**